packages/xstate-react/src/useSelector.ts

```diff
@@ -10,10 +10,15 @@ function defaultCompare<T>(a: T, b: T) {
   return a === b;
 }
 
-export function useSelector<TActor extends AnyActorRef | undefined, T>(
+export function useSelector<
+  TActor extends Pick<AnyActorRef, 'subscribe' | 'getSnapshot'> | undefined,
```

Comment:
why is the previous version not compatible?

Reply:
Because `Store` is not fully compatible with `AnyActorRef`

Reply:
Hm, ok - lots of things are missing from this. The solution is OK 👍 I would do the same in all of the other integration packages though. `@xstate/react` shouldn't be the only one that becomes compatible with this

Reply:
Agreed